supplier_app/views.py

```diff
@@ -32,10 +35,70 @@
 from . import TAXPAYER_STATUS_AP
 
 
-class SupplierHome(LoginRequiredMixin, TemplateView):
-    model = TaxPayerArgentina
+class SupplierWithoutCompanyMixin(UserPassesTestMixin):
+
+    def dispatch(self, request, *args, **kwargs):
+        if not request.user.is_authenticated():
+            self.login_url = '/'
+            return self.handle_no_permission()
+        user_test_result = self.get_test_func()()
+        if not user_test_result:
+            self.login_url = reverse('company')
+            return self.handle_no_permission()
+        return super().dispatch(request, *args, **kwargs)
+
+    def test_func(self):
+        user_company = self.request.user.companyuserpermission_set.all()
+        return len(user_company) > 0
+
+
+class CompanySelectorView(LoginRequiredMixin, CreateView):
+    model = CompanyUserPermission
+    fields = ['company']
+    template_name = 'supplier_app/company_selector.html'
+
+    def get_success_url(self):
+        return reverse('supplier-home')
+
+    def form_invalid(self, form):
+        return HttpResponseRedirect(reverse_lazy('company'), status=422)
+
+    def form_valid(self, form):
+        """
+        If the form is valid, redirect to the supplied URL.
+        """
+        companyuserpermission = form.save(commit=False)
+        companyuserpermission.user = self.request.user
+        companyuserpermission.save()
```

Comment:
I'm not sure, but I think you have an instance of the model inside `form.instance`


    def form_valid(self, form):
        form.instance.user = self.request.user
        form.save()
        ...